enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,635 @@
+                    {
+                      "id": "eno101",
+                      "type": "phy",
+                    }
+                  ],
+                  "networks": [
+                    {
+                      "id": "network0",
+                      "type": "ipv4",
+                      "link": "eno101",
+                      "routes": [
+                        {
+                          "network": "0.0.0.0",
+                          "netmask": "0.0.0.0",
+                          "gateway": "192.168.1.1"
+                        }
+                      ],
+                      "network_id": "network0"
+                    }
+                  ],
+                  "services": [
+                    {
+                      "type": "dns",
+                      "address": "8.8.8.8"
+                    },
+                  ]
+                }
+   ```
+  with means:
+   - Configure `acpiIndex: 101` on the virtual machine first interface that will have this network so we have interface name `eno101`
+   - Configure interface `eno101`, gateways, and nameservers in the OpenStack network config data (without addresses) and store at KubevirtMachineTemplate `networkData` field
+   - Pass nodepool's Network and addresses to KubevirtMachineTemplate NeworkName and Subnets
+   - Pass interface `eno101` to KubevirtMachineTemplate Interface.
+   - Create the generated KubevirtMachineTemplate.
+
+5. The Cluster API Provider KubeVirt controller reconcile the KubevirtMachineTemplate:
+   - Allocates an available IP address from the network's pool (there are different pools per network)
+   - Records the allocation in the KubevirtMachine status subresource (`allocatedBootstrapAddresses` field), so in memory pools can be re-constructed on restarts
+   - Expand the openstack network configuration with the allocated address
+   - Injects the expanded network configuration into the VM's cloud-init config drive
+
+6. When the CoreOS VM boots:
+   - Afterburn reads the OpenStack network config from the cloud-init config drive
+   - Translates the network configuration into dracut kernel arguments
+   - Applies the static network configuration to the specified interface(s) at boot before ignition
+   - Ignition starts and fetch configuration from hypershift ignition server without issues since basic network configuration is in place.
+
+7. The node comes online with the configured static IP address on the appropriate network (default or custom) and joins the hosted cluster
+
+#### Failure Handling
+
+- **IP Pool Exhaustion**: If all IPs in the pool are allocated, machine creation will fail with a clear error message indicating pool exhaustion at the NodePool CRD status with a condition like `KubevirtIPPoolExhausted=true`
+- **Invalid Network Configuration**: The API will validate network configuration format during HostedCluster creation, rejecting invalid configurations, using CEL is a perfect fit to implement this.
```

Comment:
We need to implement this at 2 levels:
- CEL first as a api-machinery protection layer
- NetworkValidations at Hypershift level, in order to avoid conflicts regarding internal rules